devito/finite_differences/differentiable.py

```diff
@@ -623,6 +623,23 @@ class Mod(DifferentiableOp, sympy.Mod):
     __sympy_class__ = sympy.Mod
 
 
+class SafeInv(Differentiable, sympy.core.function.Application):
```

Comment:
I'm now thinking that this is generated at `_evaluate()`, hence it doesn't (or better maybe, _shouldn't_) need to be a `Differentiable`. It might just have to inherit from sympy.Application and, as such, be moved into `symbolics/extended_sympy.py`

Reply:
No, this is part of `_eval_at` not `_evaluate` so the result needs to be evaluable and differentiable as it could be inside a derivative  as well.